addons/payment_adyen/models/payment.py

```diff
@@ -27,6 +168,18 @@ class AcquirerAdyen(models.Model):
     adyen_skin_code = fields.Char('Skin Code', required_if_provider='adyen', groups='base.group_user')
     adyen_skin_hmac_key = fields.Char('Skin HMAC Key', required_if_provider='adyen', groups='base.group_user')
 
+    @api.model
+    def _adyen_convert_amount(self, amount, currency):
+        """
+        Adyen requires the amount to be multiplied by 10^k,
+        where k depends on the currency code.
+        """
+        if currency.name not in CURRENCY_CODE_MAPS:
```

Comment:
Let's not bother with this. This could trigger a 500 error on the frontend while the end user couldn't do anything about it.